### ISSUE TITLE
Strange `java.lang.ArrayIndexOutOfBoundsException`

### ISSUE THREAD

**Issue by AlexWayfer** [jruby/jruby#6355]
**Environment Information**
  
Provide at least:
* JRuby version: [`jruby 9.2.12.0 (2.5.7) 2020-07-01 db01a49ba6 OpenJDK 64-Bit Server VM 25.265-b01 on 1.8.0_265-b01 +jit [linux-x86_64]`](https://cirrus-ci.com/task/5091436225888256?command=bundle#L3)
* Operating system and platform: `jruby:latest` Docker image in Cirrus CI.

Other relevant info you may wish to add:
* Installed or activated gems
* Application/framework version (e.g. Rails, Sinatra)
* Environment variables

**Expected Behavior**

Specs without fails.

**Actual Behavior**

https://cirrus-ci.com/task/5091436225888256?command=test#L2217-L2241

```
java.lang.ArrayIndexOutOfBoundsException: -1
	at org.jruby.runtime.ThreadContext.getCurrentFrame(ThreadContext.java:541)
	at org.jruby.runtime.ThreadContext.getFrameName(ThreadContext.java:675)
	at org.jruby.exceptions.RaiseException.doCallEventHook(RaiseException.java:171)
	at org.jruby.exceptions.RaiseException.preRaise(RaiseException.java:111)
	at org.jruby.exceptions.RaiseException.<init>(RaiseException.java:65)
	at org.jruby.exceptions.Exception.<init>(Exception.java:38)
	at org.jruby.exceptions.StandardError.<init>(StandardError.java:38)
	at org.jruby.exceptions.SystemCallError.<init>(SystemCallError.java:38)
	at org.jruby.RubySystemCallError.constructThrowable(RubySystemCallError.java:133)
	at org.jruby.RubyException.toThrowable(RubyException.java:349)
	at org.jruby.exceptions.RaiseException.from(RaiseException.java:81)
	at org.jruby.Ruby.newRaiseException(Ruby.java:4120)
	at org.jruby.Ruby.newErrnoFromInt(Ruby.java:3746)
	at org.jruby.Ruby.newErrnoFromInt(Ruby.java:3765)
	at org.jruby.RubyProcess.raiseErrnoIfSet(RubyProcess.java:1698)
	at org.jruby.RubyProcess.checkErrno(RubyProcess.java:1688)
	at org.jruby.RubyProcess.checkErrno(RubyProcess.java:1682)
	at org.jruby.RubyProcess.access$000(RubyProcess.java:83)
	at org.jruby.RubyProcess$3.call(RubyProcess.java:1451)
	at org.jruby.runtime.CallBlock.call(CallBlock.java:76)
	at org.jruby.runtime.Block.call(Block.java:139)
	at org.jruby.RubyProc.call(RubyProc.java:318)
	at org.jruby.internal.runtime.RubyRunnable.run(RubyRunnable.java:105)
	at java.lang.Thread.run(Thread.java:748)
```

**Comment by headius** [comment]
This is likely the same issue as #6337 which was fixed for JRuby 9.2.13 by #6323.

**Comment by AlexWayfer** [comment]
> This is likely the same issue as #6337

OK, sorry, didn't find it.
